### PR TITLE
Add discovery.serverset component

### LINKED ISSUE #2281: Flow: discovery.serverset component
For feature parity, we should bring over the upstream Prometheus SD mechanism and introduce it as a Flow component.

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#serverset_sd_config
https://github.com/twitter/finagle/tree/develop/finagle-serversets

### PR DESCRIPTION
**Pull request by thampiotr**
#### PR Description

Adds `discovery.serverset` component.

#### Which issue(s) this PR fixes

Fixes #2281

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Comment by thampiotr** [comment]
I'll be offline for most of next week, so if by some miracle there are not too many comments (or they can be pushed directly to docs), then feel free to merge this during my absence 👍  🙏

File: docs/sources/flow/reference/components/discovery.serverset.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+## Usage
+
+```river
+discovery.serverset "LABEL" {
+	servers = SERVERS_LIST
+	paths   = ZK_PATHS_LIST
+}
+```
+
+Serverset data stored in Zookeeper must be in the JSON format, the Thrift format is not currently supported.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.serverset.md:24]:
```suggestion
Serverset data stored in Zookeeper must be in JSON format. The Thrift format is not supported.
```

File: docs/sources/flow/reference/components/discovery.serverset.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Arguments
+
+The following arguments are supported:
+
+| Name      | Type           | Description                                      | Default | Required |
+|-----------|----------------|--------------------------------------------------|---------|----------|
+| `servers` | `list(string)` | Zookeeper servers to connect to.                 |         | yes      |
+| `paths`   | `list(string)` | The Zookeeper paths to discover Serversets from. |         | yes      |
+| `timeout` | `duration`     | Zookeeper session timeout                        | `10s`   | no       |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.serverset.md:34]:
```suggestion
| `servers` | `list(string)` | The Zookeeper servers to connect to.                 |         | yes      |
| `paths`   | `list(string)` | The Zookeeper paths to discover Serversets from. |         | yes      |
| `timeout` | `duration`     | The Zookeeper session timeout                        | `10s`   | no       |
```

**Review by clayton-cornell — APPROVED** [review]
Minor tweaks to the docs.

File: docs/sources/flow/reference/components/discovery.serverset.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+## Usage
+
+```river
+discovery.serverset "LABEL" {
+	servers = SERVERS_LIST
+	paths   = ZK_PATHS_LIST
+}
+```
+
+Serverset data stored in Zookeeper must be in JSON format. The Thrift format is not supported.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.serverset.md:24]:
```suggestion
Serverset data is stored in Zookeeper and must be in JSON format. The Thrift format is not supported.
```

**Review by clayton-cornell — COMMENTED** [review]
Some minor updates for the docs.

File: docs/sources/flow/reference/components/discovery.serverset.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+[Serversets]: https://github.com/twitter/finagle/tree/develop/finagle-serversets
+[Finagle]: https://twitter.github.io/finagle/
+[Aurora]: https://aurora.apache.org/
+
+## Usage
+
+```river
+discovery.serverset "LABEL" {
+	servers = SERVERS_LIST
+	paths   = ZK_PATHS_LIST
```
Comment (rfratto) [inline docs/sources/flow/reference/components/discovery.serverset.md:20]: I don't know if ZK is a well-known prefix but I found it confusing for a second; probably better to spell it out:

```suggestion
	paths   = ZOOKEEPER_PATHS_LIST
```

**Review by rfratto — APPROVED** [review]
Thanks Piotr!

**Pull request merged**